pkg/apis/channels/v1alpha1/subscription_validation.go

```diff
@@ -20,21 +20,29 @@ import (
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/knative/pkg/apis"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
 )
 
 func (s *Subscription) Validate() *apis.FieldError {
 	return s.Spec.Validate().ViaField("spec")
 }
 
 func (ss *SubscriptionSpec) Validate() *apis.FieldError {
-	if ss.Channel == "" {
-		fe := apis.ErrMissingField("channel")
-		fe.Details = "the Subscription must reference a Channel"
+	if ss.From == nil || equality.Semantic.DeepEqual(ss.From, &corev1.ObjectReference{}) {
+		fe := apis.ErrMissingField("from")
+		fe.Details = "the Subscription must reference a from channel"
```

Comment:
This should probably also ensure that the `ss.From.Name` and `ss.From.Kind` (at least) are non-empty.

(In general, it feels like there should be a function to validate an `ObjectReference` in the context of Subscription.)

Reply:
le sigh :) Yes, there's bunch of more work to be done here :) Thanks for the comments, I just don't want to keep churning these things until we settle on the shape and the names. I'm going to add validation for now that it can only be a Channel for now as discussed in Slack yesterday, and later on we can relax.